apps/definition_accept/test/accept_test.exs

```diff
@@ -0,0 +1,41 @@
+defmodule AcceptTest do
+    use ExUnit.Case
+    use Placebo
+
+    describe "new/1" do
+
+        def input do
+            %{
+                version: 1,
+                dataset_id: "123",
+                subset_id: "456",
+                destination: FakeDestination.new!(),
+                connection: FakeAcceptConnection.new!(),
+            }
+        end
+
+        test "accepts an input that contains an id" do
+            input_with_id = Map.merge(input(), %{id: "an id"})
+            assert {:ok, _} = Accept.new(input_with_id)
+        end
+
+        test "accepts an input that does not contain an id" do
+            assert {:ok, _} = Accept.new(input())
+        end
+
+        test "generates an id if not present" do
+            allow UUID.uuid4(), return: "123-456"
+
+            {:ok, observed} = Accept.new(input())
+            assert observed.id == "123-456"
+        end
+
+        test "overwrites existing id with generated one" do
+            allow UUID.uuid4(), return: "123-456"
+
+            input_with_id = Map.merge(input(), %{id: "an id"})
+            {:ok, observed} = Accept.new(input_with_id)
+            assert observed.id == "123-456"
+        end
```

Comment:
It might be nice to have a test that creates the expected struct and then does a pinned assert to make sure we are getting the struct we expect with the fields we expect.

Reply:
I reorganized the existing tests to make use of the setup feature of ExUnit, and I think it does what you are requesting. Check it out.

Reply:
Looks good! You may need to pin output though so he doesn't always match.